iceoryx_posh/test/moduletests/test_compatability_check_level.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_utils/log/logging.hpp"
+#include "iceoryx_utils/log/logstream.hpp"
+#include "mocks/logger_mock.hpp"
+
+#include "test.hpp"
+
+#include <cstdint>
+#include <limits>
+
+using namespace ::testing;
+using namespace iox::version;
+
+class CompatibilityCheckLevel_test : public Test
+{
+  public:
+    void TearDown()
+    {
+        m_loggerMock.m_logs.clear();
+    }
+
+    Logger_Mock m_loggerMock;
+};
+
+TEST_F(CompatibilityCheckLevel_test, OffLeadsToCorrectString)
+{
+    auto sut = CompatibilityCheckLevel::OFF;
```

Comment:
sadly we also can do stuff like:
```cpp
logstream << static_cast<CompatibilityCheckLevel>(18237812);
```

And this should always be translated into something like `CompatibilityCheckLevel::UNDEFINED` or something similar. I think we have to care of this in every enum to string translation function that after the switch case always the undefined string is returned.

Reply:
Thanks for spotting this! Added a default branch to the `MonitoringMode` streaming operator as well.